src/rpc/rawtransaction.cpp

```diff
@@ -1770,6 +1770,9 @@ static RPCHelpMan joinpsbts()
         if (!DecodeBase64PSBT(psbtx, txs[i].get_str(), error)) {
             throw JSONRPCError(RPC_DESERIALIZATION_ERROR, strprintf("TX decode failed %s", error));
         }
+        if (psbtx.GetVersion() != 0) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "joinpsbts only operates on version 0 PSBTs");
+        }
```

Comment:
Is there a reason for this restriction?

Reply:
IIRC updating this RPC would require pretty significant changes to it that felt out of scope for this PR. I've left it as something to do in a followup in order to keep the scope of the PR limited to just the bare minimum to work with PSBTv2.